core/ts/Poll.ts

```diff
@@ -121,33 +122,29 @@ export class Poll implements IPoll {
     maciStateRef: MaciState,
     stateTreeDepth: number,
   ) {
-    this.duration = duration;
     this.pollEndTimestamp = pollEndTimestamp;
     this.coordinatorKeypair = coordinatorKeypair;
     this.treeDepths = treeDepths;
     this.batchSizes = batchSizes;
     this.maxValues = maxValues;
     this.maciStateRef = maciStateRef;
     this.pollId = maciStateRef.polls.length;
-    this.numSignUps = Number(maciStateRef.numSignUps.toString());
     this.stateTreeDepth = stateTreeDepth;
```

Comment:
I wonder why `stateTreeDepth` is needed in `Poll`?

Seems we could remove like you removed `numSignUps`, given we can just grab that from `maciStateRef` when needed?

Reply:
good shout

Reply:
updated